src/bin/ion/commands/beta/count.rs

```diff
@@ -1,39 +1,42 @@
+use crate::commands::{IonCliCommand, WithIonCliArgument};
 use anyhow::{Context, Result};
-use clap::{Arg, ArgAction, ArgMatches, Command};
+use clap::{ArgMatches, Command};
 use ion_rs::*;
 use std::fs::File;
 use std::io::{stdin, BufReader, StdinLock};
 
-pub fn app() -> Command {
-    Command::new("count")
-        .about("Prints the number of top-level values found in the input stream.")
-        .arg(
-            // All argv entries after the program name (argv[0])
-            // and any `clap`-managed options are considered input files.
-            Arg::new("input")
-                .index(1)
-                .help("Input file [default: STDIN]")
-                .action(ArgAction::Append)
-                .trailing_var_arg(true),
-        )
-}
+pub struct CountCommand;
```

Comment:
🗺️ This PR leans into the "command" vs "namespace" dichotomy that has been weakly alluded to in prior commits. In the tree of available `ion-cli` operations, intermediate nodes are _namespaces_--groups of commands--while leaf nodes are commands.

Each command and namespace in the CLI has been given a zero-sized type like the one above (`pub struct CountCommand`) that implements the new `IonCliCommand` trait.